### PR TITLE
cheque-status field

### PR DESCRIPTION
**Pull request by rathorevaibhav**
Changes include:

1. Added a cheque_status field for invoice that is active when payment type is cheque

### CODE REVIEW

File: app/Http/Requests/Finance/InvoiceRequest.php [hunk 1]
```diff
@@ -33,6 +33,7 @@ public function rules()
             'paid_on' => 'nullable',
             'paid_amount' => 'nullable|numeric',
             'payment_type' => 'nullable|string',
+            'cheque_status' => 'nullable|string|required_if:payment_type,cheque',
```
Comment (nishanth-kd) [inline app/Http/Requests/Finance/InvoiceRequest.php]: +1

File: resources/assets/js/app.js [hunk 2]
```diff
@@ -110,3 +110,11 @@ $('.status-close').on('click', function(){
     let wrapper = $(this).closest('.alert');
     wrapper.fadeOut(500);
 });
+
+$('.form-invoice').on('change', '#payment_type', function(){
```
Comment (nishanth-kd) [inline resources/assets/js/app.js]: would suggest you to use Vue over Jquery for this kind of presentation logic. Way more leaner and cleaner.

File: resources/assets/js/app.js [hunk 3]
```diff
@@ -110,3 +110,11 @@ $('.status-close').on('click', function(){
     let wrapper = $(this).closest('.alert');
     wrapper.fadeOut(500);
 });
+
```
Comment (nishanth-kd) [inline resources/assets/js/app.js]: is there a component based JS or everything goes into main.js?

Comment (rathorevaibhav) [reply resources/assets/js/app.js]: It's just app.js right now. I'm planning to make it component based very soon. I think having a look at MMN codebase will help.

Comment (nishanth-kd) [reply resources/assets/js/app.js]: Umm I would suggest it to be a bit different from MMN. It should be isolated based on features, kike hr.js and finance.js. I think MMN currently follows a screen based thing much like how illuma does it.

**Pull request merged**